pkg/client/cli/intercept_state.go

```diff
@@ -148,7 +148,7 @@ func (ii *interceptInfo) intercept(cmd *cobra.Command, args []string) error {
 	// We don't particularly care about errors here because if we do
 	// error then we'll assume they aren't in an air-gapped environment
 	// and if there are issues with the connector, it will fail later
-	_ = ii.withConnector(true, func(cs *connectorState) error {
+	_ = ii.withConnector(false, func(cs *connectorState) error {
```

Comment:
This is very expensive. Especially for the case when an intercept is done in a disconnected state. Unless a connect has been done in advance, this will start the connector and daemon, do the call, and then kill them both. To avoid that, I suggest the following:

1. Always perform the check if the user is required to log in but isn't (it's cheap) but don't return the error. Instead, assign it to an specific error variable.
2. Defer the call to `CanConnectAmbassadorCloud()` to a point where the connector is enabled (on line 178 and 185) and don't call at all unless the error is set.

Reply:
Ah thank you Thomas! This is what was bugging me last night bc it was expensive but I was over-tired and figured I should sleep on it and return to it in the morning.